File: internal/handler/api/presenter/topic.go
package presenter

import "github.com/sparkymat/currents/internal/dbx"

type Topic struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
}

type MediaItemTopic struct {
	Topic

	Confirmed bool `json:"confirmed"`
}

func TopicFromModel(m dbx.Topic) Topic {
	t := Topic{
		ID:          m.ID.String(),
		Name:        m.Name,
		Description: m.Description,
	}

	return t
}

func MediaItemTopicFromModel(t dbx.Topic, m dbx.MediaItemTopic) MediaItemTopic {
	pt := MediaItemTopic{
		Topic:     TopicFromModel(t),
		Confirmed: m.ConfirmedAt.Valid,
	}

	return pt
}

File: frontend/src/models/Topic.ts
export class Topic {
  public id: string;

  public name: string;

  public description: string;

  public keywords: string[];

  public confirmed: boolean;

  constructor(json: any) {
    this.id = json.id;
    this.name = json.name;
    this.description = json.description;
    this.keywords = json.keywords;
  }
}
